relayer/src/chain/counterparty.rs

```diff
@@ -425,26 +426,18 @@ pub fn unreceived_acknowledgements_sequences(
 pub fn unreceived_packets(
     chain: &impl ChainHandle,
     counterparty_chain: &impl ChainHandle,
-    channel: &IdentifiedChannelEnd,
-) -> Result<Vec<u64>, Error> {
-    let counterparty = channel.channel_end.counterparty();
-    let counterparty_channel_id = counterparty
-        .channel_id
-        .as_ref()
-        .ok_or_else(Error::missing_counterparty_channel_id)?;
-
-    let (commit_sequences, _) = commitments_on_chain(
+    path: &PathIdentifiers,
+) -> Result<(Vec<u64>, Height), Error> {
```

Comment:
Can we update the doc comment for this method to detail the new return type? It would also be helpful if the doc comment explained why the `Height` is returned in the `Ok` case.

Reply:
Perhaps even introduce a struct in place of the tuple, and a newtype for the `u64` while we are at it?

Reply:
> Can we update the doc comment for this method to detail the new return type? It would also be helpful if the doc comment explained why the `Height` is returned in the `Ok` case.

fixed

> Perhaps even introduce a struct in place of the tuple, and a newtype for the `u64` while we are at it?

Maybe we do it in a different PR?

Reply:
Agree let's do it in a separate PR.
I started working on this but it wasn't as easy as it sounds. Two complications are that:

1. A struct already exists that could do the job, but that needs to be moved from `relayer-cli` to `relayer` crate and some adaptations are necessary
https://github.com/informalsystems/ibc-rs/blob/f7cbe1cfea7f4b7aeb1bf0e99a1791f46abb2e89/relayer-cli/src/commands/query/packet/acks.rs#L16-L19

2. the method `unreceived_packets` has multiple uses and we have adapt all callers if we change the API.